Doc/library/typing.rst

```diff
@@ -2010,9 +2069,12 @@ These are not used in annotations. They are building blocks for declaring types.
 
 .. class:: NewType(name, tp)
 
-   A helper class to indicate a distinct type to a typechecker,
-   see :ref:`distinct`. At runtime it returns an object that returns
-   its argument when called.
+   Helper class to create low-overhead :ref:`distinct types <distinct>`.
+
+   ``NewType`` creates an object that is considered a distinct type by a
```

Comment:
This wording with "an object" sounds a bit awkward to me, but not sure I have a better suggestion. Possibly:

"A ``NewType`` is considered a distinct type by a typechecker.
At runtime, however, calling a ``NewType`` returns the argument unchanged."

Reply:
That sounds better to me, I'll go with something like that